pkg/manager/namespace.go

```diff
@@ -32,36 +31,33 @@ const (
 )
 
 func ConfigureWatchNamespaces(options *manager.Options, log logr.Logger) {
-	namespaces := lookupEnv()
-	var watchNamespaces []string
+	namespaces := splitNamespaces(os.Getenv(WatchNamespaceEnvVar))
+
+	var namespaceConfigs map[string]cache.Config
 	if len(namespaces) != 0 {
 		log.Info("watching namespaces", "namespaces", namespaces)
-		watchNamespaces = namespaces
+		namespaceConfigs = make(map[string]cache.Config)
+		for _, namespace := range namespaces {
+			namespaceConfigs[namespace] = cache.Config{}
+		}
 	} else {
 		log.Info("watching all namespaces")
-		watchNamespaces = []string{v1.NamespaceAll}
 	}
 
 	options.NewCache = func(config *rest.Config, opts cache.Options) (cache.Cache, error) {
 		return cache.New(config, cache.Options{
-			Namespaces: watchNamespaces,
+			DefaultNamespaces: namespaceConfigs,
 		})
 	}
```

Comment:
The more idiomatic way to do this with controller-runtime now is (no need to set `options.NewCache` anymore):

```go
options.Cache.DefaultNamespaces = namespaceConfigs
```

This change breaks the tests in `namespace_test.go` (which expect to use `opts.NewCache`), but that can be easily overcome too. Wherever we check the opts.NewCache function is not nil, call it, wait for it to sync, etc. we can do this instead:
```go
	By("creating the manager")
	mgr, err := manager.New(cfg, opts)
	Expect(err).To(BeNil())

	By("starting the manager")
	ctx, cancel := context.WithCancel(context.Background())
	var wg sync.WaitGroup
	wg.Add(1)
	go func() {
		Expect(mgr.Start(ctx)).To(Succeed())
		wg.Done()
	}()

	By("waiting for the cache to sync")
	c := mgr.GetCache()
	Expect(c.WaitForCacheSync(ctx)).To(BeTrue())
```

Reply:
That setup should be more resilient to future controller-runtime API changes too since it no longer has expectations about the specific way that the options are configured.

Reply:
Fixed in latest force push